var/spack/repos/builtin/packages/flux-core/package.py

```diff
@@ -94,6 +94,7 @@ class FluxCore(AutotoolsPackage):
     # Use of distutils in configure script dropped in v0.55
     depends_on("python@:3.11", when="@:0.54", type=("build", "link", "run"))
     depends_on("py-cffi@1.1:", type=("build", "run"))
+    depends_on("py-cffi@1.16:", type=("build", "run"), when="^python@3.12:")
```

Comment:
This line isn't necessary, Older cffi can't build with newer python anyway due to the depends_on in cffi.